dspace-api/src/main/java/org/dspace/importer/external/service/DoiCheck.java

```diff
@@ -0,0 +1,52 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.importer.external.service;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+
+/**
+ * Utility class that provides methods to check if a given string is a DOI and exists on CrossRef services
+ *
+ * @author Corrado Lombardi (corrado.lombardi at 4science.it)
+ */
+public class DoiCheck {
```

Comment:
Could we add a Unit Test for this class please?  Then we could verify that all the various DOI patterns hardcoded in this class are correct...as we can add tests to verify each returns `isDoi` = true.

Also, I think the JavaDocs need minor updates, as this class doesn't appear to check if the DOI "exists on CrossRef".  I don't see any code that contacts CrossRef.

Reply:
@Micheleboychuk : Again, I think this class should have a corresponding Unit Test class to verify that both `isDoi()` and `purgeDoiValue()` are working as expected for common values of each.  I also noticed these two methods are missing JavaDocs.

Reply:
At a guess, PATTERN only matches prefixes known to be issued by CrossRef?  Which means something different than "exists on CrossRef".

The DOI IdentifierProvider already provides a method to test whether some string is a DOI, if that would be useful.

It might be a good idea to externalize these hardcoded URIs and regexes so that they can be quickly fixed if requirements change.  See #3287, #3354.

Reply:
Moved this feedback over to https://github.com/DSpace/DSpace/pull/8237#pullrequestreview-972552327, as that PR is the closest to being merged and this code exists in both these PRs.